src/DataFrame/Operations/Merge.hs

```diff
@@ -27,18 +25,18 @@ instance Semigroup D.DataFrame where
                     pure $ D.insertColumn name col df
                 | otherwise =
                     let
-                        numColumnsA = (fst $ D.dimensions a')
-                        numColumnsB = (fst $ D.dimensions b')
-                        numColumns = max numColumnsA numColumnsB
+                        numRowsA = fst $ D.dimensions a'
```

Comment:
Based on how this binding was used, I realized this and the one below were named incorrectly - `fst . dimensions` represents number of rows.

Reply:
Good catch. Another part of the code I need to revisit.